server/src/lib/model/goals.ts

```diff
@@ -349,16 +349,16 @@ export async function checkGoalsAfterContribution(
         VALUES (?,     ?,    ?,         ?,         NOW()     )`,
       [threshold, type, client_id, localeId]
     );
+  }
 
-    // reached a 3-day streak, and try to earn the three_day_streak bonus
-    let earned = false;
-    if (challenge && type === 'streak' && currentCount === 3) {
-      earned = await earnBonus('three_day_streak', [
-        client_id,
-        client_id,
-        challenge,
-      ]);
-    }
-    return earned;
+  // reached a 3-day streak, and try to earn the three_day_streak bonus
+  let earned = false;
+  if (challenge && streak_days === 3) {
+    earned = await earnBonus('three_day_streak', [
+      client_id,
+      client_id,
+      challenge,
+    ]);
   }
+  return earned;
```

Comment:
The achievement requirement is:

> "Set a goal and complete a three day streak"

This looks good to me, but I'm going to start a discussion in tonight's meeting to make sure this is the behaviour we're looking for.